pkg/framework/guest.go

```diff
@@ -238,16 +239,20 @@ func (g *Guest) WaitForNodesUp(numberOfNodes int) error {
 		}
 
 		if len(nodes.Items) != numberOfNodes {
-			return microerror.Maskf(waitError, "worker nodes are still not found")
+			return microerror.Maskf(waitError, "found %d k8s nodes but want %d", len(nodes.Items), numberOfNodes)
 		}
 
+		var notReadyCnt int
 		for _, n := range nodes.Items {
 			for _, c := range n.Status.Conditions {
 				if c.Type == v1.NodeReady && c.Status != v1.ConditionTrue {
-					return microerror.Maskf(waitError, "worker nodes are still not ready")
```

Comment:
This log in particular was very misleading. There is possibility that we have more nodes than expected in the k8s api e.g. when we scale down in draining test.

Reply:
I think we use something like `WaitForNodesDown` when scaling down. There is another helper for that IIRC.